src/StockServiceManager.php

```diff
@@ -29,13 +30,30 @@ class StockServiceManager implements StockServiceManagerInterface, StockTransact
   protected $configFactory;
 
   /**
+   * The current store.
+   *
+   * @var \Drupal\commerce_store\Entity\Store
+   */
+  protected $currentStore;
+
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $currentUser;
+
+  /**
    * Constructs a StockServiceManager object.
    *
    * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
    *   The config factory.
    */
   public function __construct(ConfigFactoryInterface $config_factory) {
     $this->configFactory = $config_factory;
```

Comment:
This doesn't look reliable to me (may have been resolved) returns the default store wen editing a product that is designated to another store . It is added to support a getContext() needed by the field widget to find its context. We should consider the field widget can only support single store sites.